BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -43,53 +57,502 @@ public struct AttributedLabel: Element, Hashable {
+
+            attributedText = mutableString
+        }
+
+        override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+            guard let first = touches.first else { return }
+            let touchedLinks = links(at: first.location(in: self))
+            trackingLinks = touchedLinks
+            applyLinkColors(activeLinks: touchedLinks)
+        }
+
+        override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?) {
+            guard let first = touches.first, let trackingLinks = trackingLinks else { return }
+            let touchedLinks = links(at: first.location(in: self))
+            let activeLinks = Set(touchedLinks).intersection(trackingLinks)
+            applyLinkColors(activeLinks: Array(activeLinks))
+        }
+
+        override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
+            guard let first = touches.first, let trackingLinks = trackingLinks else { return }
+            let touchedLinks = links(at: first.location(in: self))
+            let activeLinks = Set(touchedLinks).intersection(trackingLinks)
+            for link in activeLinks {
+                linkHandler?.onTap(link: link.text)
+            }
+
+            self.trackingLinks = nil
+            applyLinkColors()
+        }
+
+        override func touchesCancelled(_ touches: Set<UITouch>, with event: UIEvent?) {
+            trackingLinks = nil
+            applyLinkColors()
+        }
+    }
+
+}
+
+extension AttributedLabel {
+    struct Link: Equatable, Hashable {
+        var text: String
+        var range: NSRange
+    }
+
+    private final class LinkAccessibilityElement: UIAccessibilityElement {
+        private let link: AttributedLabel.Link
+        private var container: LabelView? { accessibilityContainer as? LabelView }
+
+        init(
+            container: LabelView,
+            label: String,
+            link: AttributedLabel.Link
+        ) {
+            self.link = link
+            super.init(accessibilityContainer: container)
+            accessibilityLabel = label
+        }
+
+        override var accessibilityFrameInContainerSpace: CGRect {
+            set { fatalError() }
+            get {
+                guard let container = container,
+                      let textStorage = container.makeTextStorage(),
+                      let layoutManager = textStorage.layoutManagers.first,
+                      let textContainer = layoutManager.textContainers.first
+                else {
+                    return .zero
+                }
+
+                let glyphRange = layoutManager.glyphRange(forCharacterRange: link.range, actualCharacterRange: nil)
+                return layoutManager.boundingRect(forGlyphRange: glyphRange, in: textContainer)
+            }
+        }
+
+        override func accessibilityActivate() -> Bool {
+            container?.linkHandler?.onTap(link: link.text)
+            return true
+        }
+    }
+}
+
+// MARK: Environment
```

Comment:
nit: Can we move some of this stuff to separate files?

Reply:
- [ ] Move to separate file